### PR TITLE
Fix MacOS app signing & notarization

### PR DESCRIPTION
**Pull request by Ironicbay**
Closes https://github.com/Scille/parsec-cloud/issues/7642

- [x] Enable hardened runtime for code signing to enable manual notarization (same state as v2)
- [x] Enable automatic notarization in electron-builder

### CODE REVIEW

File: client/electron/macOS/entitlements.plist [hunk 1]
```diff
@@ -0,0 +1,13 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!DOCTYPE plist PUBLIC "-//Apple//DTD PLIST 1.0//EN" "http://www.apple.com/DTDs/PropertyList-1.0.dtd">
+<plist version="1.0">
+<dict>
+	<!-- These are required for binaries built by PyInstaller -->
```
Comment (FirelightFlagboy) [inline client/electron/macOS/entitlements.plist:5]: Outdated comment

File: client/electron/package.js [hunk 2]
```diff
@@ -138,9 +138,14 @@ const options = {
     guid: '2f56a772-db54-4a32-b264-28c42970f684',
   },
 
+  afterSign: 'electron-builder-notarize',
```
Comment (FirelightFlagboy) [inline client/electron/package.js:141]: Should be set to `undefined` if `OPTS.sign` is false (to allow to build an app that is not signed)

File: client/package.json [hunk 3]
```diff
@@ -49,6 +49,7 @@
         "@ionic/vue-router": "^7.6.4",
         "@types/luxon": "^3.3.4",
         "axios": "^1.7.2",
+        "electron-builder-notarize": "^1.5.2",
```
Comment (Max-7) [inline client/package.json:52]: Should be a dev dependency. And I see `@types/luxon` in this list, it should also be a dev dependency.

**Review by Max-7 — CHANGES_REQUESTED** [review]

File: client/electron/macOS/entitlements.plist [hunk 4]
```diff
@@ -0,0 +1,13 @@
...
+	<key>com.apple.security.cs.allow-jit</key>
+	<true/>
+	<key>com.apple.security.cs.allow-unsigned-executable-memory</key>
+	<true/>
+	<key>com.apple.security.cs.disable-library-validation</key>
+	<true/>
```
Comment (FirelightFlagboy) [inline client/electron/macOS/entitlements.plist:11]: Could we remove that setting ?

The tutorial under: https://kilianvalkhof.com/2019/electron/notarizing-your-electron-application/

Don't use it

**Review by FirelightFlagboy — CHANGES_REQUESTED** [review]

File: client/electron/package.js (hunk 2 above)
Comment (FirelightFlagboy) [inline client/electron/package.js:141]: Should only be set if on darwin platform

**Review by FirelightFlagboy — APPROVED** [review]

**Review by Max-7 — APPROVED** [review]

**Pull request merged**